### PR TITLE
feat(build): add script to generate flow types

### PR DESCRIPTION
**Pull request by lesliecodes**
### Summary:
Add `./generateFlowTypes.sh` as part of `build:types` (ts + flow)

### Test Plan:
```
➜ components (ch124021) ✗ ./generateFlowTypes.sh
Generating flowtype for dist/Heading/Heading
Parsing Heading
Generating flowtype for dist/Heading/index
Parsing index
Generating flowtype for dist/Text/index
Parsing index
Generating flowtype for dist/Text/Text
Parsing Text
Generating flowtype for dist/index
Parsing index
Generating flowtype for dist/playground/button
Parsing button
Generating flowtype for dist/styles/global
Parsing global
Generating flowtype for dist/util/typography
Parsing typography
```
image

### CODE REVIEW

**Comment by lesliecodes** [comment]
This also generates files for index.d.ts (not sure if we want to explicitly exclude). Seems ok to leave for now?

**Comment by dcwither** [comment]
> ## [size-limit](https://github.com/ai/size-limit) report
> Path	Size
> packages/components/dist/index.js, packages/components/dist/Heading/Heading.js, packages/components/dist/Heading/index.js, packages/components/dist/Text/Text.js, packages/components/dist/Text/index.js, packages/components/dist/styles/global.js, packages/components/dist/util/typography.js	0 B (-100%)
> packages/components/dist/index.flow.js, packages/components/dist/index.js, packages/components/dist/Heading/Heading.flow.js, packages/components/dist/Heading/Heading.js, packages/components/dist/Heading/index.flow.js, packages/components/dist/Heading/index.js, packages/components/dist/Text/Text.flow.js, packages/components/dist/Text/Text.js, packages/components/dist/Text/index.flow.js, packages/components/dist/Text/index.js, packages/components/dist/styles/global.flow.js, packages/components/dist/styles/global.js, packages/components/dist/util/typography.flow.js, packages/components/dist/util/typography.js	6.45 KB (+100% 🔺)

I think updating `package.json` size limit check to ignore `.flow.js` files would be good (not sure about the exact glob that'll work best).

```json
"size-limit": [
    {
      "path": "packages/components/dist/**/*.js", <---
      "limit": "70KB",
      "webpack": false
    }
  ],
```

File: packages/components/generateFlowTypes.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+for i in $(find dist/* -name "*.d.ts"); do
+    [ -f "$i" ] || break
+    NAME=${i%%.*}
+    echo "Generating flowtype for $NAME"
+    node_modules/.bin/flowgen $i -o $NAME.flow.js
```
Comment (dcwither) [inline packages/components/generateFlowTypes.sh:7]: does `npx flowgen $i -o $NAME.flow.js` work?

File: packages/components/package.json [hunk 2]
```diff
@@ -44,6 +44,7 @@
   },
   "dependencies": {
     "@chanzuckerberg/eds-tokens": "^0.0.1-alpha.9",
+    "flowgen": "^1.13.0",
```
Comment (dcwither) [inline packages/components/package.json:47]: move to dev dependencies

File: packages/components/generateFlowTypes.sh (hunk 1 above)
Comment (lesliecodes) [reply packages/components/generateFlowTypes.sh:7]: yep - I'll change it to that

**Review by dcwither — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**